modules/setupadBidAdapter.js

```diff
@@ -0,0 +1,420 @@
+
+        bidResponse.ad = ad;
+        bidResponse.adUrl = adUrl;
+        bidResponses.push(bidResponse);
+      });
+    });
+
+    return bidResponses;
+  },
+
+  getUserSyncs: function (syncOptions, responses, gdprConsent, uspConsent) {
+    if (!responses?.length) return [];
+
+    const syncs = [];
+    const bidders = getBidders(responses);
+
+    if (syncOptions.iframeEnabled && bidders) {
+      const queryParams = [];
+
+      queryParams.push(`bidders=${bidders}`);
+      queryParams.push('gdpr=' + +gdprConsent.gdprApplies);
+      queryParams.push('gdpr_consent=' + gdprConsent.consentString);
+      queryParams.push('usp_consent=' + (uspConsent || ''));
+
+      const strQueryParams = queryParams.join('&');
+
+      syncs.push({
+        type: 'iframe',
+        url: SYNC_ENDPOINT + strQueryParams + '&type=iframe',
+      });
+
+      return syncs;
+    }
+
+    return [];
+  },
+
+  getPixelUrl: function (eventName, bid, timestamp) {
+    let bidder = bid.bidder || bid.bidderCode;
+    const auctionId = bid.auctionId;
+    if (bidder !== BIDDER_CODE) return;
+
+    let params;
+    if (bid.params) {
+      params = Array.isArray(bid.params) ? bid.params : [bid.params];
+    } else {
+      if (Array.isArray(bid.bids)) {
+        params = bid.bids.map((singleBid) => singleBid.params);
+      }
+    }
+
+    if (!params.length) return;
+
+    const placementIdsArray = [];
+    params.forEach((param) => {
+      if (!param.placement_id) return;
+      placementIdsArray.push(param.placement_id);
+    });
+
+    const placementIds =
+      (placementIdsArray.length && placementIdsArray.join(';')) || '';
+
+    if (!placementIds) return;
+
+    let extraBidParams = '';
+
+    if (eventName === CONSTANTS.EVENTS.BID_RESPONSE) {
+      bidder = JSON.stringify(biddersCpms);
+
+      // Add extra parameters
+      extraBidParams = `&currency=${bid.originalCurrency}`;
+    }
+
+    if (eventName === CONSTANTS.EVENTS.BID_WON) {
+      // Iterate through all bidders to find the winning bidder by using creativeId as identification
+      for (const bidderName in biddersCreativeIds) {
+        if (
+          biddersCreativeIds.hasOwnProperty(bidderName) &&
+          biddersCreativeIds[bidderName] === bid.creativeId
+        ) {
+          bidder = bidderName;
+          break; // Exit the loop if a match is found
+        }
+      }
+
+      // Add extra parameters
+      extraBidParams = `&cpm=${bid.originalCpm}&currency=${bid.originalCurrency}`;
+    }
+
+    const url = `${REPORT_ENDPOINT}?event=${eventName}&bidder=${bidder}&placementIds=${placementIds}&auctionId=${auctionId}${extraBidParams}&timestamp=${timestamp}`;
+
+    return url;
+  },
+};
+
+function getBidders(serverResponse) {
+  const bidders = serverResponse
+    .map((res) => Object.keys(res.body.ext.responsetimemillis || []))
+    .flat(1);
+
+  if (bidders.length) {
+    return encodeURIComponent(JSON.stringify([...new Set(bidders)]));
+  }
+}
+
+function getAdEl(bid) {
```

Comment:
none of the below is allowed, what are you trying to do?

Reply:
copy that, the function was removed.